### DISCUSSION TITLE
Documentation on writing type hints for fixtures/pytest methods?

### DISCUSSION THREAD

**Discussion by erichang-bcad** [pytest-dev/pytest#8290] [Q&A]
Hello,

Are there any documentations on the proper ways of writing type hints for various fixtures or pytest methods?  For example, this is what one of my `conftest.py` currently looks like:

```python
import pytest
from _pytest.config.argparsing import Parser
from _pytest.fixtures import FixtureRequest


def pytest_addoption(parser: Parser) -> None:
    parser.addoption(
        "--gen_output",
        action="store_true",
        default=False,
        help="save output files",
    )


@pytest.fixture(scope="session")
def gen_output(request: FixtureRequest) -> bool:
    return request.config.getoption("gen_output")
```

I couldn't find documentation on the proper classes to use as type hints of the `parser` and `request` variables, so I just use the `type()` method.  Of course, this is not ideal as I have to import from a pytest internal library.

So I have two questions:  Is there any documentations on the proper type hints for various pytest methods or pytest special variables?  And if not, is there any plans to expose these classes through public API so users don't have to import from pytest's internal library?  Thanks.

**Comment by bluetech** [comment]
Hopefully #7469 would answer your question!
